lib/compilers/helpers/module-name-mapper-helper.js

```diff
@@ -0,0 +1,39 @@
+const localResolve = require('./local-resolve-helper')
+
+/**
+ * Applies the moduleNameMapper substitution from the jest config
+ *
+ * @param {String} source - the original string
+ * @param {String} filePath - the path of the current file (where the source originates)
+ * @param {Object} jestConfig - the jestConfig holding the moduleNameMapper settings
+ * @returns {String} path - the final path to import (including replacements via moduleNameMapper)
+ */
+module.exports = function applyModuleNameMapper (source, filePath, jestConfig = {}) {
+  // If no moduleNameMapper is present, just return the source
+  if (!jestConfig.moduleNameMapper) return source
+
+  // Extract the moduleNameMapper settings from the jest config. TODO: In case of development via babel@7, somehow the jestConfig.moduleNameMapper might end up being an Array. After a proper upgrade to babel@7 we should probably fix this.
```

Comment:
This comment is good because it gives some information about why we check that moduleNameMapper is an array